core/src/main/java/io/methvin/watcher/DirectoryWatcher.java

```diff
@@ -105,7 +136,7 @@ public DirectoryWatcher build() throws IOException {
       if (logger == null) {
         staticLogger();
       }
-      return new DirectoryWatcher(paths, listener, watchService, enableFileHashing, logger);
+      return new DirectoryWatcher(paths, files, listener, watchService, enableFileHashing, logger);
```

Comment:
What if we just support regular files in `paths`? Is the goal just to make the API more explicit?

Reply:
Yes, I think having an explicit API helps reasoning about the overhead of watching independent files.
  
My main motivation to make a distinction between files and directories is that the user of this API needs to split the paths to watch in directories and files. To do that, the files/dirs need to exist, and therefore by construction the user filters out non existing directories (in OSX, I have to do this in bloop because the watcher will fail with an exception if a non-existing directory is passed -- the check is in `WatchablePath` IIRC).